### DISCUSSION TITLE
How to release AvaloniaRuntimeXamlLoader.Load(path) loaded object

### DISCUSSION THREAD

**Discussion by arit1988** [AvaloniaUI/Avalonia#17627] [Q&A]
### Describe the bug

When I use AvaloniaRuntimeXamlLoader.Load(path) to load an axaml file, which is a usercontrol. 
Every time called AvaloniaRuntimeXamlLoader.Load(path), the data in memory will increase. If I no longer need the obtained object and want to free up this resource, how should I do it

### To Reproduce

1 using MemoryStream memoryStream = new MemoryStream(bytes);
   var content = (UserControl)AvaloniaRuntimeXamlLoader.Load(memoryStream);
2 Grid.children.Add(content)
3 Grid.Children.Clear()

### Avalonia version

11.0.10

### OS

Windows, Linux

**Comment by maxkatz6** [comment]
You don't need to release these objects.
Garbage collector is one to do that.

**Reply by rabbitism** [reply]
But TypeBuilder will hold the parsed types unless you explicitly set it as collectable. Obviously we dont have control on this when using xaml loader.

**Reply by maxkatz6** [reply]
Do you mean, there is a considerable memory leak from dynamically created types?

**Reply by rabbitism** [reply]
> Do you mean, there is a considerable memory leak from dynamically created types?

https://learn.microsoft.com/en-us/dotnet/fundamentals/reflection/collectible-assemblies

Can read more here. Its hard to say if it is leakage because its by type, but not by instance.

**Reply by maxkatz6** [reply]
Dynamically created assembly (with its types) _should_ be dynamically collected when it's not needed: https://github.com/AvaloniaUI/Avalonia/blob/bed48a7bb62a2deb4e246065a093d36955b28c57/src/Markup/Avalonia.Markup.Xaml.Loader/AvaloniaXamlIlRuntimeCompiler.cs#L98C31-L99. I suppose there can be an infinity of reasons why it's not.

Either way, we can provide a custom assembly creation callback via [RuntimeXamlLoaderConfiguration](https://github.com/AvaloniaUI/Avalonia/blob/master/src/Markup/Avalonia.Markup.Xaml/RuntimeXamlLoaderConfiguration.cs#L6)
